src/NATS.Client.Core/Internal/BufferWriterExtensions.cs

```diff
@@ -1,25 +1,25 @@
 using System.Buffers;
 using System.Buffers.Text;
 using System.Runtime.CompilerServices;
-using System.Text;
 using NATS.Client.Core.Commands;
 
 namespace NATS.Client.Core.Internal;
 
 internal static class BufferWriterExtensions
 {
-    private const int MaxIntStringLength = 10; // int.MaxValue.ToString().Length
+    private const int MaxIntStringLength = 9; // https://github.com/nats-io/nats-server/blob/28a2a1000045b79927ebf6b75eecc19c1b9f1548/server/util.go#L85C8-L85C23
+    private static readonly StandardFormat MaxIntStringFormat = new('D', MaxIntStringLength);
```

Comment:
Should we reconsider this approach? Do we gain big enough/measurable performance benefit doing this vs. writing unnecessary stuff to the wire?

Reply:
Added logic in write loop to trim out the unneeded bytes